R/water_price.R

```diff
@@ -19,34 +20,44 @@
 #'   }
 #' 
 
-water_price <- function(gdx, file=NULL, level="reg", index=FALSE, index_baseyear=2005, digits=4) {
+water_price <- function(gdx, file=NULL, level="reg", weight = "value", index=FALSE, index_baseyear=2005, digits=4) {
+  
+  #cellular level
   oq_water_cell <- readGDX(gdx,"oq43_water","oq_water", format="first_found")[,,"marginal"]
-  oq_water_cell<- as.magpie(round(oq_water_cell,digits))
+  oq_water_cell <- as.magpie(round(oq_water_cell,digits=digits))
   
-  if(is.null(oq_water_cell)) {
-    warning("Water shadow prices cannot be calculated as needed data could not be found in GDX file! NULL is returned!")
-    return(NULL)
-  }
-
-  if(level=="cell"){
+  ovm_watdem_cell <- setNames(readGDX(gdx,"ov_watdem","ov43_watdem","ovm_watdem", format="first_found")[,,"agriculture.level"],NULL)
+  
+  #cluster level water consumption
+  weight_cell <- -oq_water_cell*ovm_watdem_cell
+  
+  
+   if(level=="cell"){
     water<- as.magpie(-oq_water_cell)
-  } else{
-    ovm_watdem_cell <- setNames(readGDX(gdx,"ov_watdem","ov43_watdem","ovm_watdem", format="first_found")[,,"agriculture.level"],NULL)
-    if(is.null(ovm_watdem_cell) ) {
-      warning("Water shadow prices cannot be calculated as needed data could not be found in GDX file! NULL is returned!")
-      return(NULL)
-    }
-    water <- as.magpie(superAggregate(as.magpie(-oq_water_cell*ovm_watdem_cell),level=level,aggr_type="sum",crop_aggr=FALSE) / superAggregate(ovm_watdem_cell,level=level,aggr_type="sum",crop_aggr=FALSE))
+  } 
+  
+  if (level!="cell") {
+    if (weight =="quantity") {
+      #regional level weight based on qty of water
+        weight_reg_qty <- superAggregate(ovm_watdem_cell,level=level,aggr_type="sum",crop_aggr=FALSE)
+        water <- as.magpie(superAggregate(as.magpie(weight_cell),level=level,aggr_type="sum",crop_aggr=FALSE) / weight_reg_qty)
+  } else if (weight == "value") {
+    #regional level weight based on value of water
+    weight_reg_value <- superAggregate(as.magpie(weight_cell),level=level,aggr_type="sum",crop_aggr=FALSE)
+    water <- as.magpie(superAggregate(as.magpie(-oq_water_cell*weight_cell),level=level,aggr_type="sum",crop_aggr=FALSE) / weight_reg_value)
     }
+  }
+  
+  
   if (index) {
     # check if the baseyear is contained in the gdx  
     if(!index_baseyear %in% getYears(water)) {
-      miss_year <- index_baseyear
       water <- time_interpolate(water, index_baseyear, integrate_interpolated_years=TRUE)
-      water <- as.magpie(round(water,digits))
-    }
+         }
     water <- water/setYears(water[,index_baseyear,],NULL)*100
   }
-  water <- as.magpie(round(water,digits))
```

Comment:
this line is actually fine. You don't have to call it with `digits=digits`, since the `round` function expects the digits at the second place of its definition.